### DISCUSSION TITLE
Change dtype of already existing zarr array

### DISCUSSION THREAD

**Discussion by aliaksei-chareshneu** [zarr-developers/zarr-python#1444] [Q&A]
Dear all,

Could you tell me please if it is possible to change the dtype of already created zarr array?

Thank you for any input,
Best regards,
Aliaksei

**Comment by d-v-b** [comment]
Hi Aliaksei,

At the moment there is no simple way to do this. The problem is that the dtype is of a zarr array is manifested in two separate places: explicitly in the .zarray metadata, and implicitly in the contents of a chunk file. If you change either one of these to a new dtype, the array will be in a broken state.

**Reply by rabernat** [reply]
One solution of this could be via an array -> array codec that transforms the dtype. This codec doesn't exist today but would be trivial to implement in the V3 framework.

Going further, it should be possible to perform any arithmetic transformation of the underlying data using this approach (e.g. rescale the data).

**Comment by aliaksei-chareshneu** [comment]
Thank you for your responses
